napari/layers/text/text.py

```diff
@@ -0,0 +1,841 @@
+    def n_dimensional(self) -> str:
+        """bool: renders points as n-dimensionsal."""
+        return self._n_dimensional
+
+    @n_dimensional.setter
+    def n_dimensional(self, n_dimensional: bool) -> None:
+        self._n_dimensional = n_dimensional
+        self.events.n_dimensional()
+        self._set_view_slice()
+
+    @property
+    def font_size(self):
+        return self._font_size
+
+    @font_size.setter
+    def font_size(self, size: float):
+        self._font_size = size
+        self.events.font_size()
+        self.events.highlight()
+
+    @property
+    def text_color(self) -> str:
+        """str: face color of marker for the next added point."""
+
+        return self._text_color
+
+    @text_color.setter
+    def text_color(self, text_color: str) -> None:
+        self._text_color = text_color
+        self.events.text_color()
+        self.events.highlight()
+
+    @property
+    def selected_data(self):
+        """list: list of currently selected points."""
+        return self._selected_data
+
+    @selected_data.setter
+    def selected_data(self, selected_data):
+        self._selected_data = list(selected_data)
+        selected = []
+        for c in self._selected_data:
+            if c in self._indices_view:
+                ind = list(self._indices_view).index(c)
+                selected.append(ind)
+        self._selected_view = selected
+        self._selected_box = self.interaction_box(self._selected_view)
+
+    def interaction_box(self, index):
+        """Create the interaction box around a list of points in view.
+
+        Parameters
+        ----------
+        index : list
+            List of points around which to construct the interaction box.
+
+        Returns
+        ----------
+        box : np.ndarray
+            4x2 array of corners of the interaction box in clockwise order
+            starting in the upper-left corner.
+        """
+        if len(index) == 0:
+            box = None
+        else:
+            data = self._data_view[index]
+            size = self.font_size
+            if data.ndim == 1:
+                data = np.expand_dims(data, axis=0)
+            data = points_to_squares(data, size)
+            box = create_box(data)
+
+        return box
+
+    @property
+    def mode(self):
+        """str: Interactive mode
+
+        Interactive mode. The normal, default mode is PAN_ZOOM, which
+        allows for normal interactivity with the canvas.
+
+        In ADD mode clicks of the cursor add points at the clicked location.
+
+        In SELECT mode the cursor can select points by clicking on them or
+        by dragging a box around them. Once selected points can be moved,
+        have their properties edited, or be deleted.
+        """
+        return str(self._mode)
+
+    @mode.setter
+    def mode(self, mode):
+        if isinstance(mode, str):
+            mode = Mode(mode)
+
+        if not self.editable:
+            mode = Mode.PAN_ZOOM
+
```

Comment:
I'd maybe put a line in here like
```python
if mode==Mode.ADD:
    mode = Mode.PAN_ZOOM
```
to prevent entering add mode for now, as i'd say that functionality can come when we have the ability to edit text too. Might also be nice to drop the add button from the GUI so its not clickable, but otherwise I'd try and keep as much as the add machinery in here that we might want to reuse.

unless we want to brainstorm supporting both adding and editing in this PR already - but I wouldn't do one without the other. I'd love it if you wanted to do both in this PR!!